### PR TITLE
refactor: hide switch dashboard option behind flag

### PR DESCRIPTION
**Pull request by mariajgrimaldi**
### Description
This PR adds a flag to turn on/off `Switch Dashboard` from the authenticated user dropdown with hard-coded references to edX. This change is temporary while maintainers determine how the switch dashboard feature can look to the community.

### Testing instructions
You can follow these instructions for the tutor setup: https://github.com/openedx/frontend-app-learner-dashboard/pull/238
But use these configurations instead: 
```
MFE_CONFIG_OVERRIDES = {
    "learner-dashboard": {
        "ENABLE_EDX_PERSONAL_DASHBOARD": False,
    }
}
```

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @mariajgrimaldi! Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


This is currently a draft pull request. When it is ready for our review and all tests are green, click "Ready for Review", or remove "WIP" from the title, as appropriate.

**Comment by itsjeyd** [comment]
Hey @mariajgrimaldi, thanks for the PR!

It looks like there are some test failures. Would you mind fixing those before engineering review starts?

**Comment by mariajgrimaldi** [comment]
@itsjeyd: of course! Thanks

**Comment by mariajgrimaldi** [comment]
@itsjeyd: this is the error message
image
I'm not sure what it means

**Comment by openedx-webhooks** [comment]
@mariajgrimaldi Even though your pull request wasn’t merged, please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/frontend-app-learner-dashboard&entry.867055334=https://github.com/openedx/frontend-app-learner-dashboard/pull/237&entry.1484655318=https://github.com/mariajgrimaldi&entry.752974735=2023-11-07+03:34&entry.1917517419=2023-11-08+17:47&entry.2133058324=No) so we can improve your experience in the future.

**Comment by arbrandes** [comment]
@mariajgrimaldi, the email error is a red herring (I'm not sure why we have this turned on in the first place).  The actual problem is just two snapshot failures in `src/containers/LearnerDashboardHeader/ExpandedHeader/AuthenticatedUserDropdown.test.jsx`.  Mind regenerating them?

**Comment by itsjeyd** [comment]
@mariajgrimaldi It looks like @arbrandes answered your question. (Thanks!) Let me know when you've had a chance to follow up on his suggestion and the build is green.

**Comment by justinhynes** [comment]
@mariajgrimaldi -- did you see the latest comment on the test failures (and possibly needing to update the snapshots)? 

Additionally, I'll be creating a backlog ticket for getting this in front of someone on our team. I could possibly prioritize getting this reviewed and merged this week if we can fix the test failures.

Thanks!

**Comment by mariajgrimaldi** [comment]
@arbrandes @justinhynes @itsjeyd: on it! Thanks

**Comment by mariajgrimaldi** [comment]
@arbrandes @justinhynes @itsjeyd: tests updated and checks green! Thank you

File: src/containers/LearnerDashboardHeader/ExpandedHeader/AuthenticatedUserDropdown.jsx [hunk 1]
```diff
@@ -29,16 +29,21 @@ export const AuthenticatedUserDropdown = () => {
...
-          </Dropdown.Item>
-          {!!dashboard && (
-            <Dropdown.Item as="a" href={dashboard.url} key={dashboard.label}>
-              {dashboard.label} {formatMessage(messages.dashboard)}
-            </Dropdown.Item>
+          { getConfig().ENABLE_EDX_PERSONAL_DASHBOARD && (
+            <>
+              <Dropdown.Header>SWITCH DASHBOARD</Dropdown.Header>
+              <Dropdown.Item as="a" href="/edx-dashboard" className="active">
+                Personal
```
Comment (justinhynes) [inline src/containers/LearnerDashboardHeader/ExpandedHeader/AuthenticatedUserDropdown.jsx:36]: It looks like lines 34 and 36 introduce some hardcoded english, was this intentional? It looks like we used to have dedicated messages/strings to support translation. Are they no longer sufficient?

Comment (mariajgrimaldi) [reply src/containers/LearnerDashboardHeader/ExpandedHeader/AuthenticatedUserDropdown.jsx:36]: Sorry. I rebased with master and completely removed the translation strings. I'll add them again

**Comment by mariajgrimaldi** [comment]
Question: should we turn off the personal dashboard by default? 🤔

**Comment by justinhynes** [comment]
If it's a 2U/edX.org feature, having it disabled by default seems totally reasonable!

**Review by justinhynes — APPROVED** [review]
LGTM

**Comment by justinhynes** [comment]
@mariajgrimaldi ugh, sorry to bug you again -- could you rebase this whenever you get a chance? 

I was going to merge it but realized this branch is out of date with `main` again.

**Comment by jsnwesson** [comment]
Hi @mariajgrimaldi ! I am taking over for Aperture's on-call this week, so I'll keep an eye on this and #238 for merging whenever you rebase them.

**Review by arbrandes — APPROVED** [review]
Looks good to me!

**Comment by justinhynes** [comment]
@mariajgrimaldi Thanks for your patience on the rebase(s). Looks good, I'll merge this shortly to ensure we don't need you to go through it again. :)

**Pull request merged**